File: pages/login_page.py
from selenium.webdriver.common.by import By
from utils.helpers import wait_for

class LoginPage:
    def __init__(self, driver):
        self.driver = driver
        self.user = (By.ID, "user-name")
        self.passw = (By.ID, "password")
        self.login_btn = (By.ID, "login-button")

    def open(self, url):
        self.driver.get(url)

    def title(self):
        return self.driver.title  # "Swag Labs"

    def login(self, username, password):
        wait_for(self.driver, self.user).clear()
        self.driver.find_element(*self.user).send_keys(username)
        self.driver.find_element(*self.passw).clear()
        self.driver.find_element(*self.passw).send_keys(password)
        self.driver.find_element(*self.login_btn).click()

File: src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage {
    private final WebDriver driver;
    private final By user = By.id("user-name");
    private final By pass = By.id("password");
    private final By loginBtn = By.id("login-button");
    private final By error = By.cssSelector("[data-test='error']");

    public LoginPage(WebDriver driver) { this.driver = driver; }
    public void open(String baseUrl) { driver.get(baseUrl); }
    public String title() { return driver.getTitle(); } // "Swag Labs"
    public void login(String u, String p) {
        driver.findElement(user).clear(); driver.findElement(user).sendKeys(u);
        driver.findElement(pass).clear(); driver.findElement(pass).sendKeys(p);
        driver.findElement(loginBtn).click();
    }
    public boolean hasError() { return !driver.findElements(error).isEmpty(); }
}
